src/cmap/message_stream.ts

```diff
@@ -165,12 +171,20 @@ function processIncomingData(stream: MessageStream, callback: Callback<Buffer>)
   let ResponseType = messageHeader.opCode === OP_MSG ? BinMsg : Response;
   if (messageHeader.opCode !== OP_COMPRESSED) {
     const messageBody = message.slice(MESSAGE_HEADER_SIZE);
-    stream.emit('message', new ResponseType(message, messageHeader, messageBody));
 
-    if (buffer.length >= 4) {
+    // If we are a monitoring message stream using the streaming protocol and
+    // there is more in the buffer that can be read, skip processing since we
+    // want the last hello command response that is in the buffer.
+    if (stream.isStreamingProtocol && buffer.length >= 4) {
       processIncomingData(stream, callback);
     } else {
-      callback();
+      stream.emit('message', new ResponseType(message, messageHeader, messageBody));
+
+      if (buffer.length >= 4) {
+        processIncomingData(stream, callback);
```

Comment:
That is correct.

Reply:
Do we currently create a new connection every time we ping the server in the polling monitor case? If not, could we get into this situation if we have a chance to send multiple requests before processing the response?

Otherwise, this would only be applicable if we implement exhaust cursor support at some point. Perhaps we should call back with an error in this case so that we can detect if something unexpected is happening.

Reply:
No we use the same connection in the polling scenario but we won't get into this scenario as the streaming protocol flag wont be set.

Reply:
This thread can be resolved from my point of view.  Even though this is dead code, it's the way it's worked for a long time and doesn't need to be changed as part of this PR.

@dariakp when your follow-up question has been been addressed please resolve the thread.

Reply:
followed up async: we only send out the next polling request after receiving a response to the previous one, so no concerns there, resolving